manifests/charts/istio-control/istio-discovery/files/kube-gateway.yaml

```
@@ -27,6 +27,7 @@ spec:
       annotations:
         {{- toJsonMap
           (omit .Annotations "kubectl.kubernetes.io/last-applied-configuration" "gateway.istio.io/name-override" "gateway.istio.io/service-account" "gateway.istio.io/controller-version")
+          (strdict "istio.io/rev" .Revision | default "default")
```

Does order of operations work here correctly? 

```suggestion
          (strdict "istio.io/rev" (.Revision | default "default"))
```

may be more explicit if not

---

I think its not since the test cases are not showing `"default"` but `""`

---

> I think its not since the test cases are not showing "default" but ""

It wasn't clear to me, because in unit tests the value of `istio.io/rev` was evaluated to `""`, but when I created a gateway on KinD, I got annotation `istio.io/rev: default`, so I was suspecting that it's a result of using fake dependencies in unit tests or something like that.

Anyway, thanks for catching it.